website/docs/r/external_network_v2.html.markdown

```diff
@@ -0,0 +1,176 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_external_network_v2"
+sidebar_current: "docs-vcd-resource-external-network-v2"
+description: |-
+  Provides a VMware Cloud Director External Network resource (version 2). New version of this resource
+  uses new VCD API and is capable of creating NSX-T backed external networks as well as port group
+  backed ones.
+---
+
+# vcd\_external\_network\_v2
+
+Provides a VMware Cloud Director External Network resource (version 2). New version of this resource 
+uses new VCD API and is capable of creating NSX-T backed external networks as well as port group
+backed ones.
+
+~> **Note:** This resource uses new VMware Cloud Director
+[OpenAPI](https://code.vmware.com/docs/11982/getting-started-with-vmware-cloud-director-openapi) and
+requires at least VCD *10.0+*. (NSX-T *3.0+* requires VCD *10.1.1+*)
+
+Supported in provider *v3.0+*.
+
+## Example Usage (NSX-T backed external network)
+
+```hcl
+data "vcd_nsxt_manager" "main" {
+  name = "nsxManager"
+}
+
+data "vcd_nsxt_tier0_router" "router" {
+  name            = "first-router"
+  nsxt_manager_id = data.vcd_nsxt_manager.main.id
+}
+
+resource "vcd_external_network_v2" "ext-net-nsxt" {
+  name        = "nsxt-network"
+  description = "First NSX-T backed network"
+
+  nsxt_network {
+    nsxt_manager_id      = data.vcd_nsxt_manager.main.id
+    nsxt_tier0_router_id = data.vcd_nsxt_tier0_router.router.id
+  }
+
+  ip_scope {
+    enabled       = false
+    gateway       = "88.88.88.1"
+    prefix_length = "24"
+
+    static_ip_pool {
+      start_address = "88.88.88.88"
+      end_address   = "88.88.88.100"
+    }
+  }
+
+  ip_scope {
+    # enabled       = true # by default
+    gateway       = "14.14.14.1"
+    prefix_length = "24"
+
+    static_ip_pool {
+      start_address = "14.14.14.10"
+      end_address   = "14.14.14.15"
+    }
+    
+    static_ip_pool {
+      start_address = "14.14.14.20"
+      end_address   = "14.14.14.25"
+    }
+  }
+}
+```
+
+## Example Usage (NSX-V backed external network)
+```hcl
+data "vcd_vcenter" "vc" {
+  name = "vc1"
+}
+
+data "vcd_portgroup" "sw" {
+  name = "TestbedPG"
+  type = "DV_PORTGROUP"
+}
+
+resource "vcd_external_network_v2" "ext-net-nsxv" {
+  name        = "TestAccVcdExternalNetworkBasic"
+  description = "Test External Network"
+
+  vsphere_network {
+    vcenter_id     = data.vcd_vcenter.vc.id
+    portgroup_id   = data.vcd_portgroup.sw.id
+  }
+
+  ip_scope {
+    gateway      = "192.168.30.49"
+    netmask      = "24"
```

Comment:
May the example be broken? I get these errors with it:

```
$ terraform apply

Error: Missing required argument

  on nsx-t.tf line 90, in resource "vcd_external_network_v2" "ext-net-nsxv":
  90:   ip_scope {

The argument "prefix_length" is required, but no definition was found.


Error: Unsupported argument

  on nsx-t.tf line 92, in resource "vcd_external_network_v2" "ext-net-nsxv":
  92:     netmask      = "24"

An argument named "netmask" is not expected here.
```

Reply:
Correct. Leftover from old example. Changed.